src/UndoCommands.cpp

```diff
@@ -292,18 +360,35 @@ redo()
 {
   _scene->clearSelection();
 
-  _newSceneJson = takeSceneJsonFromClipboard();
-
-  if (_newSceneJson.empty())
-    return;
-
-  _newSceneJson = makeNewNodeIdsInScene(_newSceneJson);
+  // Ignore if pasted in content that does not generate nodes.
+  try
+  {
+    insertSerializedItems(_newSceneJson, _scene);
+  }
+  catch(...)
+  {
```

Comment:
Do we delete all the selected nodes if the "ctrl+v" was unsuccessful ?
Sorry, did not get this branch of the code

Reply:
I think this is necessary.

In the `calculator` example, If you paste the following, using the no-delete method, you will only get the `Result` node. and it cannot be undone.

So I think it's best to delete all created content, or you can also check if all nodes are creatable first, and then go ahead and create them.

```json
{
    "connections": [
        {
            "inPortIndex": 0,
            "intNodeId": 1,
            "outNodeId": 0,
            "outPortIndex": 0
        }
    ],
    "nodes": [
        {
            "id": 1,
            "internal-data": {
                "model-name": "Result"
            },
            "position": {
                "x": 75,
                "y": -16
            }
        },
        {
            "id": 0,
            "internal-data": {
                "model-name": "N--Source",
                "number": "0"
            },
            "position": {
                "x": -258,
                "y": -83
            }
        }
    ]
}

```

Reply:
I see your point, thanks